### PR TITLE
CloudFormation support.

### PR DESCRIPTION
**Pull request by jsmartin**
This adds a AWS CloudFormation module, an example playbook, and an example CloudFormation template.

### CODE REVIEW

**Comment by mpdehaan** [comment]
Can you include an example of doing something to the AWS guests post launch similar to the EC2 module?

This would probably help people understand stack_outputs more.  I suspect it returns a data structure that can be used with add_hosts in the same way?

**Comment by jsmartin** [comment]
So stack_outputs are defined per CloudFormation template.  It theoretically could be something ingestible by add_hosts, such as a hostname, ip, etc. This would be possible if launching single instances at a time, but not using AutoScale Groups as there's not a way (that I know) to provide the set of IPs associated with the group to a CloudFormation output.

To work around this, I'm relying on the ec2.py inventory script and performing 2 different ansible playbook runs so that the inventory is updated on the second run.

I think I could provide another cloudformation example that does what you want.

**Comment by mpdehaan** [comment]
Sounds reasonable, maybe make a note of how users should use the inventory
script in that case.

**Comment by jsmartin** [comment]
I do have this verbage in the cloudformation.yaml

"#The thought here is to bring up a bare infrastructure with CloudFormation, but use ansible to configure it.
#I generally do this in 2 different playbook runs as to allow the ec2.py inventory to be updated."

**Comment by mpdehaan** [comment]
good deal, in queue for looking at tomorrow and likely merging, thanks!

File: library/cloudformation [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+  region:
+    description:
+      - The AWS region the stack will be launched in
+    required: true
+    default: null
+    aliases: []
+  state:
+    description:
+      - If state is "present", stack will be created.  If state is "present" and if stack exists and template has changed, it will be updated.
+        If state is present, stack will be removed.
```
Comment (lwade) [inline library/cloudformation:52]: If state is absent

**Comment by lwade** [comment]
Can't wait to try it out! Nice work :)

I think I found a typo, commented in the diff.

**Comment by mpdehaan** [comment]
merged, thanks, this is devel branch so we can obviously tweak as needed as we go.

It is now possible to have a EXAMPLES="""
"""

docstring in the module docs for easier formatting of documentation, so it may be useful to include something there.

**Pull request merged**